packages/cli-hydrogen/src/cli/services/deploy/config.ts

```diff
@@ -24,3 +25,19 @@ export const getDeployConfig = async (config: DeployConfig): Promise<ReqDeployCo
     path: directory,
   }
 }
+
+const insideGitDirectory = async (directory: string) => {
+  try {
+    await git.ensureInsideGitDirectory(directory)
+  } catch (err: unknown) {
+    if ((err as error.Abort)?.message !== git.OutsideGitDirectoryError().message) throw error
+
+    const shouldGitInit = await gitInit()
+
+    if (!shouldGitInit) throw new error.AbortSilent()
+
+    await git.initializeRepository(directory)
+    await git.createGitIngore(directory)
+    await git.commitAll('Initial Commit', {directory})
```

Comment:
```suggestion
    await git.commitAll('Initial commit generate by create-hydrogen', {directory})
```

(or something like that, maybe check in with Graham on some of these?)